d3a_api_client/utils.py

```diff
@@ -288,18 +287,42 @@ def log_market_progression(message):
 
 def log_bid_offer_confirmation(message):
     try:
-        if message.get("status") == "ready":
+        if message.get("status") == "ready" and message.get("command") in ["bid", "offer"]:
             event = message.get("command")
             data_dict = json.loads(message.get(event))
             energy = data_dict.get("energy")
             price = data_dict.get("price")
+            rate = price / energy
             trader = data_dict.get("seller" if event=="offer" else "buyer")
             logging.info(f"{trader} {'OFFERED' if event == 'offer' else 'BID'} "
-                         f"{round(energy, 2)} kWh at {price} cts/kWh")
+                         f"{round(energy, 2)} kWh at {rate} cts/kWh")
     except Exception as e:
         logging.error(f"Logging bid/offer info failed.{e}")
 
 
+def log_deleted_bid_offer_confirmation(message, command_type, bid_offer_id):
+    try:
+        if message.get("status") == "ready":
+            if bid_offer_id is not None:
+                logging.info(f"<-- All {command_type}s are successfully deleted-->")
+            else:
+                logging.info(f"<-- {command_type} {bid_offer_id} is successfully deleted-->")
```

Comment:
Could you please explain to me when the `bid_offer_id` can be `None`? Why does that mean that all bids/offers have been successfully deleted?

Reply:
@fievelk from the caller code 
```python
    def delete_offer(self, offer_id=None):
        transaction_id, posted = self._post_request('delete-offer', {"offer": offer_id})
        if posted:
            response = self.dispatcher.wait_for_command_response('offer_delete', transaction_id)
            log_deleted_bid_offer_confirmation(response, "offer", offer_id)
            return response
```
if the offer_id is None, the d3a will delete all the offers at once